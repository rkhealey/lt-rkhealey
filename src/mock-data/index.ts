import profile from './profile.png';

export const user = {
  username: '@yourname',
  imageSrc: profile,
};

export const theme = {
  background_color: 'rebeccapurple',
  text_color: 'palegoldenrod',
};

export const links = [
  {
    url: 'http://www.google.com?1',
    text: 'Link One',
  },
  {
    url: 'http://www.google.com?2',
    text: 'Link Two',
  },
  {
    url: 'http://www.google.com?3',
    text: 'Link Three',
  },
  {
    url: 'http://www.google.com?4',
    text: 'Link Four',
  },
  {
    url: 'http://www.google.com?5',
    text: 'Link Five',
  },
];

export const showLinks = {
  title: 'Shows',
  links: [
    {
      date: 'April 01 2019',
      venue: 'The Forum, Melbourne',
      url: 'http://www.songkick.com/1',
      available: true,
    },
    {
      date: 'April 02 2019',
      venue: 'Venue Name, Canberra',
      url: 'http://www.songkick.com/2',
      available: false,
    },
    {
      date: 'April 03 2019',
      venue: 'Venue Name, Sydney',
      url: 'http://www.songkick.com/3',
      available: true,
    },
    {
      date: 'April 04 2019',
      venue: 'Venue Name, Brisbane',
      url: 'http://www.songkick.com/4',
      available: true,
    },
  ],
};

export const musicPlayer = {
  title: 'Music',
  artist: 'John Lennon',
  name: 'Imagine',
  links: [
    {
      id: 'spotify',
      platform: 'Spotify',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'apple-music',
      platform: 'Apple Music',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&1',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'soundcloud',
      platform: 'Soundcloud',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&2',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
    {
      id: 'youtube-music',
      platform: 'Youtube Music',
      url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&3',
      songId: '7pKfPomDEeI4TPT6EOYjn9',
    },
  ],
};
