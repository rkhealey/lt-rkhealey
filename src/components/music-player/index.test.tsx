import React from 'react';
import { shallow } from 'enzyme';
import MusicPlayer from '.';

import { musicPlayer, theme } from '../../mock-data';

describe('Music Player', () => {
  it('Renders', () => {
    const wrapper = shallow(<MusicPlayer {...musicPlayer} theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
