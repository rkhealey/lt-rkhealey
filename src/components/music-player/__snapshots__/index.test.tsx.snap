// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music Player Renders 1`] = `
<Collapse
  theme={
    Object {
      "background_color": "rebeccapurple",
      "text_color": "palegoldenrod",
    }
  }
  title="Music"
>
  <styled.div
    bg="grey_100"
  >
    <Platform
      id="spotify"
      key="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ"
      name="Spotify"
      onLaunchPlayer={[Function]}
      platform="Spotify"
      songId="7pKfPomDEeI4TPT6EOYjn9"
      url="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ"
    />
    <Platform
      id="apple-music"
      key="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&1"
      name="Apple Music"
      onLaunchPlayer={[Function]}
      platform="Apple Music"
      songId="7pKfPomDEeI4TPT6EOYjn9"
      url="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&1"
    />
    <Platform
      id="soundcloud"
      key="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&2"
      name="Soundcloud"
      onLaunchPlayer={[Function]}
      platform="Soundcloud"
      songId="7pKfPomDEeI4TPT6EOYjn9"
      url="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&2"
    />
    <Platform
      id="youtube-music"
      key="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&3"
      name="Youtube Music"
      onLaunchPlayer={[Function]}
      platform="Youtube Music"
      songId="7pKfPomDEeI4TPT6EOYjn9"
      url="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ&3"
    />
  </styled.div>
</Collapse>
`;
